packages/langium/src/documents/document-builder.ts

```diff
@@ -54,8 +55,8 @@ export class DefaultDocumentBuilder implements DocumentBuilder {
         this.indexManager = services.index.IndexManager;
     }
 
-    build(document: LangiumDocument): BuildResult {
-        const parseResult = this.parser.parse(document);
+    build<T extends AstNode = AstNode>(document: LangiumDocument<T>): BuildResult<T> {
+        const parseResult = this.parser.parse(document) as ParseResult<T>;
```

Comment:
I don't think this cast is necessary here.

```suggestion
        const parseResult = this.parser.parse<T>(document);
```

Reply:
Ah yes. Thanks. The parse method just got generic in the last change, wasn't at the moment I've did the upper change.